catroid/src/main/java/org/catrobat/catroid/ui/fragment/MainFormulaEditorCategoryListFragment.java

```diff
@@ -606,28 +562,44 @@ private void closeCategoryListFragment() {
 		fragTransaction.commit();
 	}
 
-	public class FormulaEditorCategory {
+	protected class FormulaEditorCategory {
 		private int[] categoryItemIds = new int[] {};
 		private int[] categoryParameterIds = new int[] {};
 		private Map<Integer, String> headerOrderMap = new TreeMap<>();
 
-		public void addHeader(String text) {
+		boolean addCategoryIfDoesNotExist(String header, int[] items, int[] parameters) {
```

Comment:
The return value is checked multiple times in MainFormulaEditorCategoryListFragment and FormulaEditorCategoryListFragment of the Phiro flavor